Marksman/Index.fs

```diff
@@ -12,7 +12,8 @@ type Index =
       headingsBySlug: Map<Slug, list<Node<Heading>>>
       wikiLinks: array<Node<WikiLink>>
       mdLinks: array<Node<MdLink>>
-      linkDefs: array<Node<MdLinkDef>> }
+      linkDefs: array<Node<MdLinkDef>>
+      yamlFrontMatter: Node<YamlContent> option }
```

Comment:
nit: option<Node<YamlContent>>?

Although I like the cleanliness of `ty option` this is a legacy syntax inherited from OCaml and AFAIU F# community is moving away from it.

Reply:
oh good, I thought it was preferred because I saw it in Ionide sources.
I prefer `option<Node>` anyways, gives me those warm Java vibes